src/comparator.rs

```diff
@@ -0,0 +1,865 @@
+>(
+    comp: COMP,
+    input: I,
+    lower_threshold: L,
+    upper_threshold: U,
+    config: Config,
+    rcc: &mut Rcc,
+) -> WindowComparator<COMP2, COMP1, Disabled> {
+    let (comp1, comp2) = comp.split(rcc);
+    (comp2, comp1).window_comparator(input, lower_threshold, upper_threshold, config, &rcc.clocks)
+}*/
+
+#[cfg(not(any(
+    feature = "stm32g473",
+    feature = "stm32g483",
+    feature = "stm32g474",
+    feature = "stm32g484"
+)))]
+type Comparators = (COMP1, COMP2, COMP3, COMP4);
+
+#[cfg(any(
+    feature = "stm32g473",
+    feature = "stm32g483",
+    feature = "stm32g474",
+    feature = "stm32g484"
+))]
+type Comparators = (COMP1, COMP2, COMP3, COMP4, COMP5, COMP6, COMP7);
+
+/// Enables the comparator peripheral, and splits the [`COMP`] into independent [`COMP1`] and [`COMP2`]
+pub fn split(_comp: COMP, rcc: &mut Rcc) -> Comparators {
+    // Enable COMP, SYSCFG, VREFBUF clocks
+    rcc.rb.apb2enr.modify(|_, w| w.syscfgen().set_bit());
+
+    // Reset COMP, SYSCFG, VREFBUF
+    rcc.rb.apb2rstr.modify(|_, w| w.syscfgrst().set_bit());
+    rcc.rb.apb2rstr.modify(|_, w| w.syscfgrst().clear_bit());
+
+    (
+        COMP1 { _rb: PhantomData },
+        COMP2 { _rb: PhantomData },
+        COMP3 { _rb: PhantomData },
+        COMP4 { _rb: PhantomData },
+        #[cfg(any(
+            feature = "stm32g473",
+            feature = "stm32g483",
+            feature = "stm32g474",
+            feature = "stm32g484"
+        ))]
+        COMP5 { _rb: PhantomData },
+        #[cfg(any(
+            feature = "stm32g473",
+            feature = "stm32g483",
+            feature = "stm32g474",
+            feature = "stm32g484"
+        ))]
+        COMP6 { _rb: PhantomData },
+        #[cfg(any(
+            feature = "stm32g473",
+            feature = "stm32g483",
+            feature = "stm32g474",
+            feature = "stm32g484"
+        ))]
+        COMP7 { _rb: PhantomData },
+    )
+}
+
+pub trait ComparatorSplit {
+    /// Enables the comparator peripheral, and splits the [`COMP`] into independent [`COMP1`] and [`COMP2`]
+    fn split(self, rcc: &mut Rcc) -> Comparators;
+}
+
+impl ComparatorSplit for COMP {
+    fn split(self, rcc: &mut Rcc) -> Comparators {
+        split(self, rcc)
+    }
+}
+
+pub trait OutputPin<COMP> {
+    fn setup(self);
+}
+
+#[allow(unused_macros)] // TODO: add support for more devices
+macro_rules! output_pin {
+    ($COMP:ident, $pin:ident, $AF:ident, $mode_t:ident, $into:ident) => {
+        impl OutputPin<$COMP> for $pin<Output<$mode_t>> {
+            fn setup(self) {
+                self.$into::<$AF>();
+            }
+        }
+    };
+    ($($COMP:ident: $pin:ident, $AF:ident,)+) => {$(
+        output_pin!($COMP, $pin, $AF, PushPull, into_alternate);
+        output_pin!($COMP, $pin, $AF, OpenDrain, into_alternate_open_drain);
+    )+};
+}
+
+// TODO: look up alternate functions for more devices than g474
+// https://www.mouser.se/datasheet/2/389/stm32g474cb-1600828.pdf#page=73
```

Comment:
please use the documents from ST site only. also use their public id like as ANxxx, RMxxx, etc.